plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/ScanQueryPageSource.java

```diff
@@ -182,6 +197,9 @@ public Page getNextPage()
 
     public static Object getField(Map<String, Object> document, String field)
     {
+        if (document == null) {
```

Comment:
This should be a `requireNonNull()`

Reply:
If value returned from `Map<String, Object> value = (Map<String, Object>) getField(document, columnHandle.baseName());` is null which can directly be passed to `decoders.get(i).decode(hit, () -> getField(leafDocument, columnHandle.dereferencePath().getLast()), columnBuilders[i]);`